ddht/tools/lru_sql_dict.py

```diff
@@ -0,0 +1,271 @@
+import pickle
+import sqlite3
+from typing import Any, Iterator, MutableMapping, NamedTuple, Tuple, Union
+
+from eth_utils.toolz import first
+
+CREATE_CACHE_QUERY = """
+    CREATE TABLE IF NOT EXISTS cache (
+        id INTEGER NOT NULL PRIMARY KEY,
+        key BLOB NOT NULL UNIQUE,
```

Comment:
Any reason why you didn't make this the primary key?

Reply:
Not particularly, it seemed like best practice from what I read for an lru cache to have a hashmap that points to a doubly linked list - which is what I was going for here. But it does seem doable to remove the hashmap entirely and just have a single table that acts as the cache w/ `key (primary key), value, pref, nref`.